### PR TITLE
Add custom hostname to support server customer use [semver:minor]

### LINKED ISSUE #91: [semver:minor] Support Server Customer with an option to specify the hostname
**Is your feature request related to a problem? Please describe.**
Server Customer can not use this orb.

**Describe the solution you'd like**
It would be great if there would be an option to set a custom host so that server customer can a also use this orb.

**Describe alternatives you've considered**
Fork this repository and change the hostname from `circleci.com` to your own custom host, then deploy the orb within the server.

### PR DESCRIPTION
**Pull request by nanophate**
### Checklist

- [x] All new jobs, commands, executors, parameters have descriptions
- [x] Examples have been added for any significant new features
- [ ] README has been updated, if necessary

### Motivation, issues

close: #91 

This change will help server user to also use this orb on their own hosted server.

### Description

Replaced the `https://circleci.com` to use a pipeline parameters. 
Added pipeline parameter default to be `https://circleci.com` so that cloud customer would not get this impact.

### CODE REVIEW

**Review by kelvintaywl — COMMENTED** [review]
perhaps we can assume HTTPS is always to be used, so the parameter's default value can be just `circleci.com` rather than `https://circleci.com`

**Comment by eddiewebb** [comment]
Thank you @nanophate , what a great addition.

I agree with @kelvintaywl , all CCI servers should still be behind HTTPS, and it is really the hostname we're adjusting.

Also, please add:
1) a sample config file using new value (see test/inputs)
2) a simple assertion test that loads that config (which itself just does replacemernt and validation) and if possible a simple assertion the output script contains a line matching the provided header.  https://github.com/eddiewebb/circleci-queue/blob/main/test/test_expansion.bats#L26-L36

File: src/commands/until_front_of_line.yml [hunk 1]
```diff
@@ -93,7 +98,7 @@ steps:
           if [ -z "${<< parameters.circleci-api-key >>}" ]; then
             echo "<< parameters.circleci-api-key >> not set. Private projects will be inaccessible."
           else
-            fetch "https://circleci.com/api/v2/me" "/tmp/me.cci"
+            fetch "<< parameters.circleci-hostname >>/api/v2/me" "/tmp/me.cci"
             me=$(jq -e '.id' /tmp/me.cci)
             echo "Using API key for user: ${me}"
```
Comment (eddiewebb) [inline src/commands/until_front_of_line.yml]: worth updating `user: ${me}, host: << parameters.circleci-hostname >>`

File: src/commands/until_front_of_line.yml [hunk 2]
```diff
@@ -40,6 +40,11 @@ parameters:
     type: string
     default: ""
     description: "Allow multiple job names to be blocked until front of line f.ex '^runTests*'"
+  circleci-hostname:
+    type: string
+    default: "https://circleci.com"
```
Comment (eddiewebb) [inline src/commands/until_front_of_line.yml]: reduce to just host. we must assume https as best practice.

File: src/commands/until_front_of_line.yml [hunk 3]
```diff
@@ -104,15 +109,15 @@ steps:
         fetch_filtered_active_builds(){
           if [ "<<parameters.consider-branch>>" != "true" ];then
             echo "Orb parameter 'consider-branch' is false, will block previous builds on any branch." 
-            jobs_api_url_template="https://circleci.com/api/v1.1/project/${VCS_TYPE}/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}?filter=running"
+            jobs_api_url_template="<< parameters.circleci-hostname >>/api/v1.1/project/${VCS_TYPE}/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}?filter=running"
```
Comment (eddiewebb) [inline src/commands/until_front_of_line.yml]: Declare as BASH variable once near beginning (like https://github.com/eddiewebb/circleci-queue/blob/main/src/commands/until_front_of_line.yml#L47) and then use env_var thorugh script.

This was authored before support for `include` but I want to move that way ans so avoid as much inline parameters as possible for now.

`CIRCLECI_BASEURL="http://<< parameters.circleci-hostname >>"`
And that would be a great line to assert in the tests

File: src/jobs/block_workflow.yml [hunk 4]
```diff
@@ -51,3 +55,4 @@ steps:
       confidence: <<parameters.confidence>>
       circleci-api-key: <<parameters.circleci-api-key>>
       job-regex: <<parameters.job-regex>>
+      circleci-hostname: <<parameters.circleci-hostname>>
```
Comment (eddiewebb) [inline src/jobs/block_workflow.yml:58]: especially given name says `host` we should update default and script to match.

**Review by eddiewebb — CHANGES_REQUESTED** [review]
Thank you so much. A few adjustments please, and In look forward to merging this.

**Comment by eddiewebb** [comment]
bors try

**Comment by eddiewebb** [comment]
BotComment: *Development* version of orb available for manual validation - `eddiewebb/queue@dev:436`

**Review by eddiewebb — APPROVED** [review]

**Comment by eddiewebb** [comment]
bors try

**Comment by eddiewebb** [comment]
BotComment: *Development* version of orb available for manual validation - `eddiewebb/queue@dev:442`

**Comment by eddiewebb** [comment]
@nanophate - would you please try `eddiewebb/queue@dev:442` and report back your results?
CC: @kelvintaywl

**Comment by nanophate** [comment]
@eddiewebb Thank you for updating the pr by adding test and suggested changes. I tested out the orb on server and it is working as expected!! Thankyou so much for checking this out 🙇 

https://app.gke-circleci.nanophate.com/pipelines/github/nanophate-cci/check-queue/11/workflows/5118df17-510f-49fd-9703-ca2f2bb9b18f/jobs/22?invite=true#step-101-276

https://github.com/nanophate-cci/check-queue/blob/main/.circleci/config.yml

Screenshot 2023-02-13 at 11 35 50

```sh
Attempting to access CircleCI api. If the build process fails after this step, ensure your CIRCLECI_API_KEY  is set.
DEBUG: Making API Call to https://gke-circleci.nanophate.com/api/v1.1/project/github/nanophate-cci/check-queue/tree/main?filter=running
DEBUG: API Success
API access successful
Checking time of workflow: 5118df17-510f-49fd-9703-ca2f2bb9b18f
DEBUG: Making API Call to https://gke-circleci.nanophate.com/api/v2/workflow/5118df17-510f-49fd-9703-ca2f2bb9b18f
DEBUG: API Success
Workflow was created at: 2023-02-13T02:35:22Z
Checking time of workflow: 4c3d4fec-6272-4229-b305-fc02794a4b0f
DEBUG: Making API Call to https://gke-circleci.nanophate.com/api/v2/workflow/4c3d4fec-6272-4229-b305-fc02794a4b0f
DEBUG: API Success
Workflow was created at: 2023-02-13T02:35:08Z
Orb parameter block-workflow is false.
Only blocking execution if running previous jobs matching this job: deploy
Oldest job: 20
This Workflow Timestamp: "2023-02-13T02:35:22Z"
Oldest Workflow Timestamp: "2023-02-13T02:35:08Z"
This build (22) is queued, waiting for build number (20) to complete.
Total Queue time: 11 seconds.

```

**Comment by eddiewebb** [comment]
bors merge

**Comment by eddiewebb** [comment]
BotComment: *Development* version of orb available for manual validation - `eddiewebb/queue@dev:445`

**Comment by eddiewebb** [comment]
BotComment: *Production* version of orb available for use - `eddiewebb/queue@1.12.0`

**Comment by eddiewebb** [comment]
Wha the heck?  i thought this weould be 1.10...  Anyway its up as 1.12

**Pull request merged**